### PR TITLE
Make MailConfig a sub-class of NetClientOptions

### LINKED ISSUE #79: add support for proxy setting in NetClient

### PR DESCRIPTION
**Pull request by sfitts**
This allows for the setting of all of the options used to create the
connection to the SMTP server and not just the few that were exposed by
the MailConfig class.

Fixes #79

### CODE REVIEW

**Comment by gaol** [comment]
@sfitts Thank you for re sending the PR to master branch, would you please rebase to solve conflicts? :)

File: src/main/java/io/vertx/ext/mail/MailConfig.java [hunk 1]
```diff
@@ -402,9 +388,15 @@ public String getKeyStorePassword() {
...
   public MailConfig setKeyStorePassword(String keyStorePassword) {
-    this.keyStorePassword = keyStorePassword;
+    JksOptions options = getTrustStoreOptions();
+    if (options == null) {
+      options = new JksOptions();
+      this.setTrustOptions(options);
+    }
+    options.setPassword(keyStorePassword);
     return this;
   }
```
Comment (gaol) [inline src/main/java/io/vertx/ext/mail/MailConfig.java]: When we extends another `DataObject`, we need to override the `setter` methods to be able to keep it fluent. At least the `setProxyOptions` should be override in this case as it is the issue we want to address in the first place.

File: src/main/java/io/vertx/ext/mail/MailConfig.java [hunk 2]
```diff
@@ -177,6 +169,19 @@ public MailConfig(JsonObject config) {
...
+  @Override
+  public String getHostnameVerificationAlgorithm() {
+    // If not explicitly set then see if we should provide a default based on SMTP settings
+    String verification = super.getHostnameVerificationAlgorithm();
+    if ((verification == null || verification.isEmpty()) && !isTrustAll() &&
+        (isSsl() || starttls != StartTLSOptions.DISABLED)) {
+      // we can use HTTPS verification, which matches the requirements for SMTPS
+      verification = "HTTPS";
+    }
+    return verification;
```
Comment (gaol) [inline src/main/java/io/vertx/ext/mail/MailConfig.java]: I am inclined to keep the `getter` method as simple as possible, and move the logic to the pool to call `setHostnameVerificationAlgorithm('HTTPS')`

**Comment by sfitts** [comment]
Will do, thanks.

**Comment by gaol** [comment]
@sfitts any progress ? :)

**Comment by sfitts** [comment]
Sorry for the delay (was a busy month...).  Here's the update (rebased on current master).

**Comment by gaol** [comment]
@sfitts Thanks for the contributions. :) 

Added breakage description at: https://github.com/vert-x3/wiki/wiki/4.0.0-Deprecations-and-breaking-changes#%EF%B8%8F-mailconfig-extends-netclientoptions

**Pull request merged**